packages/utils/src/lib/merge-configs.unit.test.ts

```diff
@@ -0,0 +1,404 @@
+import { describe, expect, it } from 'vitest';
+import { CoreConfig, PluginConfig } from '@code-pushup/models';
+import { mergeConfigs } from './merge-configs';
+
+const MOCK_CONFIG_PERSIST = {
+  persist: {
+    outputDir: '.code-pushup/packages/app',
+    format: ['json', 'md'],
+  },
+} as CoreConfig;
+
+const MOCK_JUST_ESLINT_PLUGIN = {
+  slug: 'eslint',
+  title: 'ESLint',
+  icon: 'eslint',
+  description: 'Mock Code PushUp ESLint plugin',
+  docsUrl: 'https://www.npmjs.com/package/@code-pushup/eslint-plugin',
+  packageName: '@code-pushup/eslint-plugin',
+  version: '0.0.0',
+
+  audits: [
+    {
+      slug: 'no-const-assign',
+      title: 'Disallow reassigning `const` variables',
+      description: 'ESLint rule **no-const-assign**.',
+      docsUrl: 'https://eslint.org/docs/latest/rules/no-const-assign',
+    },
+    {
+      slug: 'no-debugger',
+      title: 'Disallow the use of `debugger`',
+      description: 'ESLint rule **no-debugger**.',
+      docsUrl: 'https://eslint.org/docs/latest/rules/no-debugger',
+    },
+  ],
+
+  groups: [
+    {
+      slug: 'problems',
+      title: 'Problems',
+      description:
+        'Code that either will cause an error or may cause confusing behavior.',
+      refs: [
+        { slug: 'no-const-assign', weight: 1 },
+        { slug: 'no-debugger', weight: 1 },
+      ],
+    },
+  ],
+
+  runner: {
+    command: 'node',
+    args: ['bin.js'],
+    outputFile: 'node_modules/.code-pushup/runner-output.json',
+  },
+} as const satisfies PluginConfig;
+
+const MOCK_CONFIG_FULL_ESLINT: CoreConfig = {
+  plugins: [MOCK_JUST_ESLINT_PLUGIN],
+};
+
+const MOCK_CONFIG_NX_VALIDATORS: CoreConfig = {
+  plugins: [{ slug: 'my-plugin' } as PluginConfig],
+  categories: [
+    {
+      slug: 'nx-validators',
+      title: 'Workspace Validation',
+      refs: [],
+    },
+  ],
+};
+
+const MOCK_CONFIG_BASIC_PLUGIN: CoreConfig = {
```

Comment:
We could move these objects into a fixture folder to reduce the loc of the file

Reply:
I personally disagree, the individual test assertions rely on knowing exactly what's in these mocks, so colocating them actually makes it easier to understand the tests for me.